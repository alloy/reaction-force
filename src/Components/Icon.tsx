// @ts-ignore
import React, { StatelessComponent } from "react"
import styled from "styled-components"
import "../Assets/Fonts"
import icons, { IconName } from "../Assets/Icons"

export type FontName = string

export interface IconProps extends React.HTMLProps<HTMLDivElement> {
  font?: FontName
  name: IconName
  color?: string
  fontSize?: string
  style?: any
  onClick?: () => void
}

const Icon: React.SFC<IconProps> = props => (
  <div className={props.className} style={props.style} onClick={props.onClick}>
    {icons[props.name]}
  </div>
)

export default styled(Icon)`
  font-family: ${props => props.font || "artsy-icons"};
  color: ${props => props.color || "purple"};
  font-size: ${props => props.fontSize || "24px"};
  margin: 0 5px;
  display: inline-block;
  letter-spacing: 0px;
`
