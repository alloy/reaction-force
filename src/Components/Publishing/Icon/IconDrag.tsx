import React, { Component } from "react"

export class IconDrag extends Component<any, null> {
  static defaultProps = {
    color: "white",
    background: "black",
  }

  render() {
    return (
      <svg
        className="IconDrag"
        x="0px"
        y="0px"
        viewBox="0 0 30 30"
        version="1.1"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g transform="translate(-1062.000000, -935.000000)">
          <g transform="translate(1062.000000, 934.000000)">
            <g transform="translate(0.785741, 0.701007)">
              <circle fill={this.props.background} cx="14.5" cy="15" r="14.5" />
              <g
                transform="translate(10.000000, 6.500000)"
                strokeWidth="2"
                stroke={this.props.color}
              >
                <polyline
                  fill={this.props.background}
                  transform="translate(4.500000, 4.923718) rotate(-45.000000) translate(-4.500000, -4.923718)"
                  points="1.5 1.92371784 7.5 1.92371784 7.5 7.92371784"
                />
                <polyline
                  fill={this.props.background}
                  transform="translate(4.500000, 12.227216) rotate(-225.000000) translate(-4.500000, -12.227216)"
                  points="1.5 9.2272156 7.5 9.2272156 7.5 15.2272156"
                />
                <path d="M4.5,1.98457492 L4.5,16.0089159" />
              </g>
            </g>
          </g>
        </g>
      </svg>
    )
  }
}
