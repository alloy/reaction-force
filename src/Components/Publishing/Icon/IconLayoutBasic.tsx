import React, { Component } from "react"

export class IconLayoutBasic extends Component<any, null> {
  render() {
    return (
      <svg
        className="layout-basic"
        width="45px"
        height="30px"
        viewBox="0 0 45 30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
          <g className="layout-basic-group" fill={this.props.fill ? this.props.fill : "#000"}>
            <polyline points="6 25 40 25 40 23 6 23" />
            <polyline points="10 30 36 30 36 28 10 28" />
            <rect x="0" y="0" width="45" height="20" />
          </g>
        </g>
      </svg>
    )
  }
}
