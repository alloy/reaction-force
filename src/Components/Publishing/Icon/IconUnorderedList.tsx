import React, { Component } from "react"

export class IconUnorderedList extends Component<any, null> {
  static defaultProps = {
    color: "black",
  }

  render() {
    return (
      <svg
        className="IconUnorderedList"
        x="0px"
        y="0px"
        viewBox="-297 389.3 15.7 15.7"
        enableBackground="new -297 389.3 15.7 15.7"
        version="1.1"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          x="-291.6"
          y="390.9"
          fill={this.props.color}
          width="10.2"
          height="1.3"
        />
        <rect
          x="-291.6"
          y="396.5"
          fill={this.props.color}
          width="10.2"
          height="1.3"
        />
        <rect
          x="-291.6"
          y="402.1"
          fill={this.props.color}
          width="10.2"
          height="1.3"
        />
        <circle cx="-295.6" cy="391.5" fill={this.props.color} r="1.4" />
        <circle cx="-295.6" cy="397.1" fill={this.props.color} r="1.4" />
        <circle cx="-295.6" cy="402.8" fill={this.props.color} r="1.4" />
      </svg>
    )
  }
}
