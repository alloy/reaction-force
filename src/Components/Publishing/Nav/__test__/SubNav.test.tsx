import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import Icon from "../../../Icon"
import { IconHamburger } from "../../Icon/IconHamburger"
import { SubNav } from "../SubNav"

describe("SubNav", () => {
  it("renders SubNav", () => {
    const buttons = renderer.create(<SubNav />)
    expect(buttons).toMatchSnapshot()
  })
  it("toggles the mobile nav", () => {
    const nav = mount(<SubNav />)
    nav.find(IconHamburger).simulate("click")
    expect(nav.state().mobileNavIsOpen).toBe(true)
    nav.find(Icon).simulate("click")
    expect(nav.state().mobileNavIsOpen).toBe(false)
  })
})
