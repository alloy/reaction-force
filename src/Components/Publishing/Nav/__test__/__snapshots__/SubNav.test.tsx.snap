// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubNav renders SubNav 1`] = `
.c4 {
  background: #e5e5e5;
  color: black;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px 30px;
  font-size: 13px;
  line-height: 1;
  outline: 0;
  -webkit-transition: background-color 0.25s,color 0.25s;
  transition: background-color 0.25s,color 0.25s;
  margin: 10px;
  border: none;
  box-sizing: border-box;
  text-decoration: none;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c4:hover:not(:disabled) {
  cursor: pointer;
  background: #cccccc;
}

.c3 {
  background: black;
  color: white;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  height: 30px;
  width: 80px;
  border-radius: 2px;
  margin: 8px;
}

.c3:hover:not(:disabled) {
  background: #6e1fff;
}

.c6 {
  background: #e5e5e5;
  color: black;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px 30px;
  font-size: 13px;
  line-height: 1;
  outline: 0;
  -webkit-transition: background-color 0.25s,color 0.25s;
  transition: background-color 0.25s,color 0.25s;
  margin: 10px;
  border: none;
  box-sizing: border-box;
  text-decoration: none;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 11px;
  line-height: 1.65em;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  height: 30px;
  width: 80px;
  background-color: white;
  border-radius: 2px;
  margin: 8px;
}

.c6:hover:not(:disabled) {
  cursor: pointer;
  background: #cccccc;
}

.c1 {
  text-decoration: none;
}

.c8 {
  display: none;
  margin-right: 5px;
  max-width: 32px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
}

@media (max-width:720px) {
  .c0 .c2,
  .c0 .c5 {
    display: none;
  }

  .c0 .c7 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div
  className="c0"
>
  <a
    className="mlh-login c1"
    href="/log_in"
  >
    <button
      className="c2 c3 c4"
    >
      <span>
        Login
      </span>
    </button>
  </a>
  <a
    className="mlh-signup c1"
    href="/sign_up"
  >
    <button
      className="c5 c6"
    >
      <span>
        Signup
      </span>
    </button>
  </a>
  <div
    className="c7 c8"
  >
    <svg
      enableBackground="new 0 0 32 32"
      fill="white"
      onClick={[Function]}
      version="1.1"
      viewBox="0 0 32 32"
      width="32px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="2"
        width="20"
        x="6"
        y="22"
      />
      <rect
        height="2"
        width="20"
        x="6"
        y="15"
      />
      <rect
        height="2"
        width="20"
        x="6"
        y="8"
      />
    </svg>
  </div>
</div>
`;
